src/test/java/library/view/ModeTest.java

```diff
@@ -0,0 +1,37 @@
+package library.view;
+
+import library.repository.BookRepository;
+import library.repository.FileBookRepository;
+import library.repository.InMemoryBookRepository;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ModeTest {
+
+    @Test
+    @DisplayName("모드 코드로 모드를 찾을 수 있어야 합니다.")
+    void testFindByCode() {
+        Optional<Mode> normalMode = Mode.findByCode("1");
+        assertEquals(Mode.NORMAL, normalMode.get());
+
+        Optional<Mode> testMode = Mode.findByCode("2");
+        assertEquals(Mode.TEST, testMode.get());
+
+        Optional<Mode> invalidMode = Mode.findByCode("3");
+        assertFalse(invalidMode.isPresent());
+    }
+
```

Comment:
```suggestion
    @ValueSource(strings = {"3", "4"})
    @ParameterizedTest(name = "모드 코드 {0}에 해당하는 모드가 없으면 Optional.empty()를 반환합니다.")
    void failFindByCodeTest(String code) {
        Optional<Mode> actual = Mode.findByCode(code);
        assertThat(actual).isEmpty();
    }

    @NullAndEmptySource
    @ParameterizedTest(name = "모드 코드 {0}에 해당하는 모드가 없으면 Optional.empty()를 반환합니다.")
    void failFindByNullAndEmptyCodeTest(String code) {
        Optional<Mode> actual = Mode.findByCode(code);
        assertThat(actual).isEmpty();
    }
```

제공하지 않는 값, 즉 로직상의 결함이 발생할 수 있는 값이 들어옴에도 이를 내부적으로 정상처리로 돌릴 수 있도록 내결함성을 검증하기 위해 별도의 테스트를 작성하는 것이 좋습니다.